### PR TITLE
Creating multiple rows in the datagrid

### PR DESCRIPTION
**Pull request by poulch**
I want to merge this change because it fixes https://github.com/saleor/saleor-dashboard/issues/2858

**PR intended to be tested with API branch:** 

### Screenshots
Screenshot 2023-01-18 at 12 00 01

### Pull Request Checklist

1. [x] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
APPS_MARKETPLACE_API_URI=https://marketplace-gray.vercel.app/api/v2/saleor-apps
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [x] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: src/components/Datagrid/Header/HeaderButtonFullScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+const useStyles = makeStyles(
+  theme => ({
+    headerBtn: {
+      marginBottom: theme.spacing(2),
+    },
+
+    fullScreenIcon: {
+      fontSize: 14,
```
Comment (Droniu) [inline src/components/Datagrid/Header/HeaderButtonFullScreen.tsx:19]: **suggestion**: use `theme.typography`

File: src/components/Datagrid/useDatagridChange.ts [hunk 2]
```diff
@@ -127,11 +128,23 @@ function useDatagridChange(
...
 
-  const onRowAdded = useCallback(() => {
-    const newAdded = [...added, rows - removed.length + added.length];
-    setAdded(newAdded);
-    notify(changes.current, newAdded, removed);
-  }, [added, notify, removed, rows]);
+  const onRowAdded = useCallback(
+    (rowAmount: number = 1) => {
+      const getArrayByRowAmount = (rowAmount: number) =>
+        Array.from({ length: rowAmount }, (_, i) => i);
```
Comment (Droniu) [inline src/components/Datagrid/useDatagridChange.ts:134]: **suggestion:** this is nice, but super confusing to me
```suggestion
        [...Array(rowAmount).keys()]
```

or 
```suggestion
        Array.from(Array(rownAmount).keys())
```

File: src/components/Datagrid/useDatagridChange.ts [hunk 3]
```diff
@@ -127,11 +128,23 @@ function useDatagridChange(
...
+
+      const lastAddedIndex = rows - removed.length + added.length;
+
+      const newAdded = [
+        ...added,
+        ...map(getArrayByRowAmount(rowAmount), count => lastAddedIndex + count),
```
Comment (Droniu) [inline src/components/Datagrid/useDatagridChange.ts:140]: **question**: anyway why lodash map and not es6 map?

File: src/components/Datagrid/useDatagridChange.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { updateAtIndex } from "@dashboard/utils/lists";
 import { EditableGridCell, Item } from "@glideapps/glide-data-grid";
+import { map } from "lodash-es";
```
Comment (Droniu) [inline src/components/Datagrid/useDatagridChange.ts:3]: **nitpick:** vite probably does tree-shaking anyway so this wouldn't make a difference, but just to be sure I'd import lodash packages like this
```suggestion
import map from "lodash/map";
```

**Review by Droniu — DISMISSED** [review]

**Review by Droniu — APPROVED** [review]

File: src/components/Datagrid/useDatagridChange.test.tsx [hunk 5]
```diff
@@ -39,6 +39,19 @@ describe("useDatagridChange", () => {
...
+
+    act(() => result.current.onRowAdded(5));
+
+    expect(result.current.added).toHaveLength(5);
+    expect(result.current.added[0]).toBe(10);
+    expect(result.current.added[1]).toBe(11);
+    expect(result.current.added[2]).toBe(12);
+    expect(result.current.added[3]).toBe(13);
+    expect(result.current.added[4]).toBe(14);
+  });
```
Comment (andrzejewsky) [inline src/components/Datagrid/useDatagridChange.test.tsx:53]: arrange, act, assert

**Review by andrzejewsky — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Pull request closed without merging**